packages/dds/tree/src/feature-libraries/modular-schema/modularChangeTypes.ts

```diff
@@ -56,7 +56,14 @@ export interface ModularChangeset extends HasFieldChanges {
 	 */
 	readonly nodeAliases: ChangeAtomIdBTree<NodeId>;
 	readonly crossFieldKeys: CrossFieldKeyTable;
-	readonly constraintViolationCount?: number;
+	/**
+	 * The number of constraint violations that apply to the input context of the changeset, i.e., before this change is applied.
+	 */
+	readonly inputConstraintViolationCount?: number;
+	/**
+	 * The number of constraint violations that apply to the output context of the changeset, i.e., after this change is applied.
+	 */
+	readonly outputConstraintViolationCount?: number;
```

Comment:
"outputConstraintViolationCount" and the description make it sound as though this constraint is will prevent the transaction (as opposed to the undo) from being applied if this count is >0. I think we probably want to use the word "revert" instead of "output".

Reply:
Renamed it to use the word "inverse". I see that "revert" and "invert" are used interchangeably. I chose "invert" because the function`invert` is where this gets converted to a regular constraint.